### ISSUE TITLE
Contain numItems, nodeSize and possibly array type in data buffer?

### ISSUE THREAD

**Issue by mourner** [mourner/flatbush#12]
Currently, transferring and recreating the index is a bit cumbersome because of the need to also transfer other parameters associated with the index:

```js
self.postMessage({
  numItems: index.numItems,
  nodeSize: index.nodeSize,
  data: index.data
}, [index.data]);
...
const index = new FlatBush(e.data.numItems, e.data.nodeSize, Int32Array, e.data.data);
```

What if we include `numItems` and `nodeSize` to the data buffer so they don't need to be transferred? We can also go further and include a numeric indicator of the typed array, so that we only need an array buffer and nothing else to recreate the full index:

```js
self.postMessage(index.data, [index.data]);
...
const index = FlatBush.from(e.data);
```

The number of items and nodeSize will have to be `Uint32Array` because we can't guess the index type from the buffer alone. We would probably put them at the start of `data`. The type would be a number from 0 to 8.

@tyrasd what do you think, worth it?

**Comment by tyrasd** [comment]
Great idea.

What about also adding a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) at the start of the buffer, so that it can be checked if the data is actually valid flatbush data? This could for example also help prevent/debug issues of this kind: the data format is changed a bit in the future, but people still try to load in data generated by an older flatbush version (think: backend code not updated to the same flatbush version as the frontend)?

**Comment by mourner** [comment]
Released as v3.0.0.
